packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-typescript-visitor.ts

```diff
@@ -70,6 +70,30 @@ export class AppSyncModelTypeScriptVisitor<
     return enumDeclarations.string;
   }
 
+  protected getOwnerFieldName(modelObj: CodeGenModel): string | undefined {
+    let rule = this.getOwnerAuthRule(modelObj);
+    if (rule === undefined) {
+      return undefined
+    }
+    let ownerField = rule.ownerField
+    if (ownerField === undefined) {
+      ownerField = "owner"
+    }
+    return ownerField
+  }
+
+  protected getOwnerAuthRule(modelObj: CodeGenModel): any {
+    for (let directive of modelObj.directives) {
+      if ("auth" === directive.name) {
+        for (let rule of directive.arguments.rules) {
+          if ("owner" === rule.allow) {
+            return rule;
+          }
```

Comment:
`@auth` directive supports multiple owner rules, for instance this is an valid auth rule
```gql
type Todo
@model
@auth(
  rules: [
    { allow: owner, ownerField: "ownedBy" }
    { allow: owner, ownerField: "owner" }
  ]
) {
  id: ID!
  name: String!
  description: String
}
```